src/Agent.Listener/Agent.cs

```diff
@@ -10,102 +10,121 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener
     [ServiceLocator(Default = typeof(Agent))]
     public interface IAgent : IAgentService
     {
+        CancellationTokenSource TokenSource { get; set; }
         Task<int> ExecuteCommand(CommandLineParser parser);
     }
 
     public sealed class Agent : AgentService, IAgent
     {
+        public CancellationTokenSource TokenSource { get; set; }
+
         private Guid _sessionId;
         private int _poolId;
         private ITerminal _term;
+        private bool _inConfigStage;
 
         public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
             _term = HostContext.GetService<ITerminal>();
-        }
+        }        
 
         public async Task<int> ExecuteCommand(CommandLineParser parser)
         {
-            // TODO Unit test to cover this logic
-            Trace.Info(nameof(ExecuteCommand));
-            var configManager = HostContext.GetService<IConfigurationManager>();
+            try
+            {
+                _inConfigStage = true;
+                _term.CancelKeyPress += CtrlCHandler;
+                // TODO Unit test to cover this logic
+                Trace.Info(nameof(ExecuteCommand));
+                var configManager = HostContext.GetService<IConfigurationManager>();
 
-            // command is not required, if no command it just starts and/or configures if not configured
+                // command is not required, if no command it just starts and/or configures if not configured
 
-            // TODO: Invalid config prints usage
+                // TODO: Invalid config prints usage
 
-            if (parser.Flags.Contains("help"))
-            {
-                PrintUsage();
-                return 0;
-            }
+                if (parser.Flags.Contains("help"))
```

Comment:
looks like your file merge result get something weird. ex: you have 2 parser.IsCommand("unconfigure") and the order of those if code block get changed unnecessary.